src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -23,6 +24,16 @@ static string GetRoutingKey(Type type, string key = "")
 
             var implementedInterface = interfaces.FirstOrDefault();
 
+            if (interfaces.Count > 0)
+            {
+                Logger.Warn("The direct routing topology might not deliver message types with interfaces in their inheritance hierarchy. Message types used for publishing should not implement interfaces.");
+
+                if (interfaces.Count > 1)
+                {
+                    Logger.ErrorFormat("Publishing type has multiple interfaces. Interfaces are {0}, the one that will be used for building routing key is {1}", string.Join(", ", interfaces), implementedInterface);
```

Comment:
Regarding the log level, I still think it should be a warning. Someone could have this all running right now with no problems. They upgrade, and all of a sudden error level logs are being spewed everywhere, but no error exists. I think warning level is correct, since it indicates a _possible_ error.